tensorflow/contrib/solvers/python/ops/linear_equations.py

```diff
@@ -88,17 +91,20 @@ def cg_step(i, state):
     alpha = state.gamma / util.dot(state.p, z)
     x = state.x + alpha * state.p
     r = state.r - alpha * z
-    q = preconditioner.apply(r)
-    gamma = util.dot(r, q)
-    beta = gamma / state.gamma
-    p = q + beta * state.p
+    if preconditioner is None:
+      gamma = util.l2norm_squared(r)
```

Comment:
This code was originally written before we had tf.linalg.norm. Could you please change it to use that function now?

Reply:
shall we just call `matmul()` instead of `util.dot()`?